### PR TITLE
Whats new in Camel 4.4 WIP

### PR DESCRIPTION
**Pull request by davsclaus**
Please help update this docs, and fill in the TODO and add other important stuff you think is worth to mention.

We may want to add a bit about stuff around pipe and eventing for Camel K that this team helped with.

### CODE REVIEW

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+You can now more easily run `camel-jbang` with custom `log4j2.properties` file to use your logging configuration,
+instead of the _built-in_ logging.
+
+Added support for using Jolokia 2.x with `camel-jbang`.
+
+## Kamelets
+
+When using Kamelets then the _route snippets_ due not use any error handler. This means that when calling a Kamelet
+then if any error happens these are thrown back, and allows to use your current error handling configuration. This
+avoids any confusing, and you can regard calling a Kamelet just as calling a component; If they fail the exception is thrown back.
```
Comment (squakez) [inline content/blog/2024/02/camel44-whatsnew/index.md:107]:
```suggestion
avoids any confusion, and you can regard calling a Kamelet just as calling a component; If they fail the exception is thrown back.
```

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+making it easy to build examples and prototypes that can easily be customized to users need, when trying.
+
+You can now more easily run `camel-jbang` with custom `log4j2.properties` file to use your logging configuration,
+instead of the _built-in_ logging.
+
+Added support for using Jolokia 2.x with `camel-jbang`.
+
+## Kamelets
+
+When using Kamelets then the _route snippets_ due not use any error handler. This means that when calling a Kamelet
```
Comment (apupier) [inline content/blog/2024/02/camel44-whatsnew/index.md:105]:
```suggestion
When using Kamelets then the _route snippets_ do not use any error handler. This means that when calling a Kamelet
```

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+TODO: Throttler EIP
+
+### Variables
+
+The biggest new feature in Camel 4.4 is the introduction of [variables](/manual/variables.adoc).
+
+A variable is a key/value that can hold a value that can either be private per `Exchange`, or shared per route, or globally.
+
+With variables, you can now more easily share data between routes and/or globally. Variables is also readily accessible
```
[the remark this answers is not in the document]
Comment (davsclaus) [reply content/blog/2024/02/camel44-whatsnew/index.md:34]: Not here in the blog - but on the actual doc of the feature itself - the blog is just high level

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+## Camel Core
+
+The simple language has been improved with `hash` function, and further improved the embedded functions for `jsonpath`, `jq` and `xpath`
+making it easier to grab data from JSon or XML within your simple expression or predicates.
+
+We have optimized data formats to avoid converting payload to `byte[]` when unmarshalling, but allowing each data format
+to unmarshal the payload _as-is_. 
+
+Added JMX operation to update routes at runtime via XML format (requires turning on this feature). This can be
```
[the remark this answers is not in the document]
Comment (davsclaus) [reply content/blog/2024/02/camel44-whatsnew/index.md:21]: its documented in the MBeans operations, and their javadoc. And in general in the JMX docs.

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+that allows to use `camel-jbang` to manage and operate Camel K integrations.
+
+You can now run `camel-jbang` with `--prompt` that lets users type in placeholder values during startup,
+making it easy to build examples and prototypes that can easily be customized to users need, when trying.
+
+You can now more easily run `camel-jbang` with custom `log4j2.properties` file to use your logging configuration,
+instead of the _built-in_ logging.
+
+Added support for using Jolokia 2.x with `camel-jbang`.
+
```
Comment (tadayosi) [inline content/blog/2024/02/camel44-whatsnew/index.md:102]: Thanks! Can we mention here that `camel hawtio` now defaults to Hawtio 3.0.1?

Comment (davsclaus) [reply content/blog/2024/02/camel44-whatsnew/index.md:102]: Yes good idea

**Comment by davsclaus** [comment]
@iMashtak do you want to add something about the new kotlin-api DSL for this blog? Then you are welcome to add to the PR or add comment here for us to put into the blog

**Review by orpiske — APPROVED** [review]
LGTM, thanks @davsclaus!

Yesterday I added 2 brief notes about the `camel-kafka` changes and the Throttle EIP improvements. 
Today I included a note about fixes for `camel-kafka`, a brief overview of the internal changes on `camel-core` and corrected just a few grammar issues.

**Comment by iMashtak** [comment]
@davsclaus I think the following will be enough:

---

We have introduced new experimental API for defining Camel routes in Kotlin. It provides [Kotlin type-safe builders](https://kotlinlang.org/docs/type-safe-builders.html) as building blocks of Camel entities.

Kotlin API is not the replacement of Kotlin DSL - it is a separate project for now. While it is experimental there may be breaking changes in API in next releases.

See more in the `[docs](there must be link to documentation of camel-kotlin-api component)`.

---

I didnt find link of documentation of component, sadly. Also i dont think that there should be any repetition of docs to make release note less verbose. I included main parts from my point of view.

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+### Variables
+
+The biggest new feature in Camel 4.4 is the introduction of [variables](/manual/variables.adoc).
+
+A variable is a key/value that can hold a value that can either be private per `Exchange`, or shared per route, or per Camel context.
+
+With variables, you can now more easily share data between routes and/or globally. Variables are also readily accessible
+from EIPs and languages, just as message _headers_ or _exchange properties_ are. In other words, they have been added
+as first-class into Camel.
```
Comment (gzurowski) [inline content/blog/2024/02/camel44-whatsnew/index.md:40]:
```suggestion
as first-class primitives into Camel.
```

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+You can find a small example here:
+
+- https://github.com/apache/camel-kamelets-examples/tree/main/jbang/variables
+
+We have also made a selected number of EIPs have _special use_ of variables. The idea is to make it easier to
```
Comment (gzurowski) [inline content/blog/2024/02/camel44-whatsnew/index.md:46]:
```suggestion
We also made a selected number of EIPs to have _special use_ of variables. The idea is to make it easier to
```

File: content/blog/2024/02/camel44-whatsnew/index.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+Here you can see the `alc2` variable is set from a `jq` expression that grabs the alcohol field, and remove the `%` sign.
+The source (input) is from another variable named `beer2`. You can also specify `header:myHeader` to refer to a header, or
+if you remove `source` then the message body is used as input (default).
+The result is converted to a Java `float`,
+
+**NOTE:** 
+If you take a closer look in the screenshot above, then the debugger reveals that the `Exchange` has no message headers,
+and that the message body is empty (`null`). The example is exclusively using variables to gather and compute data; without
+the need to store and restore data on message header/body during routing. This is a cleaner and elegant practice.
```
Comment (gzurowski) [inline content/blog/2024/02/camel44-whatsnew/index.md:90]:
```suggestion
the need to store and restore data on message header/body during routing. This is a cleaner and more elegant practice.
```

**Review by gzurowski — APPROVED** [review]

**Pull request merged**